### PR TITLE
Reset selected item only when list values change

### PR DESCRIPTION
**Pull request by lujiajing1126**
Fix issue #22

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 build
 node_modules
 yarn.lock
+.vscode
```
Comment (sindresorhus) [inline .gitignore:4]: This should be in your own global gitignore, not here. https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

File: package.json [hunk 2]
```diff
@@ -7,8 +7,14 @@
...
 		"email": "vdemedes@gmail.com",
-		"url": "github.com/vadimdemedes"
+		"url": "https://github.com/vadimdemedes"
 	},
+	"contributors": [
+		{
+			"name": "Megrez Lu",
+			"email": "lujiajing1126@gmail.com"
+		}
+	],
```
Comment (sindresorhus) [inline package.json:17]: The `contributors` field is not used by npm or any tool and it's a chore to maintain. Your contribution will be reflected in the commit history and https://github.com/vadimdemedes/ink-select-input/graphs/contributors

File: .gitignore (hunk 1 above)
Comment (lujiajing1126) [reply .gitignore]: Thanks for metioning this! Fixed

File: package.json (hunk 2 above)
Comment (lujiajing1126) [reply package.json]: I see. Removed

File: src/SelectInput.tsx [hunk 3]
```diff
@@ -82,7 +83,7 @@ function SelectInput<V>({
 	const previousItems = useRef<Array<Item<V>>>(items);
 
 	useEffect(() => {
-		if (!isEqual(previousItems.current, items)) {
+		if (!isEqual(_map(previousItems.current, 'value'), _map(items, 'value'))) {
```
Comment (vadimdemedes) [inline src/SelectInput.tsx:86]: Could you use native `.map()` function over the one in lodash?

File: package.json [hunk 4]
```diff
@@ -7,7 +7,7 @@
 	"author": {
 		"name": "Vadim Demedes",
 		"email": "vdemedes@gmail.com",
-		"url": "github.com/vadimdemedes"
+		"url": "https://github.com/vadimdemedes"
```
Comment (vadimdemedes) [inline package.json:10]: Unrelated change, could you revert it?

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,3 @@
 build
 node_modules
-yarn.lock
+yarn.lock
```
Comment (vadimdemedes) [inline .gitignore:3]: Unrelated change, could you revert it?

**Review by vadimdemedes — CHANGES_REQUESTED** [review]

**Comment by vadimdemedes** [comment]
Minor changes, but looks good overall, thanks!

File: package.json (hunk 4 above)
Comment (lujiajing1126) [reply package.json]: Reverted

File: .gitignore (hunk 5 above)
Comment (lujiajing1126) [reply .gitignore]: Reverted

File: src/SelectInput.tsx (hunk 3 above)
Comment (lujiajing1126) [reply src/SelectInput.tsx]: Changed. But there seems to be a style issue.

Comment (lujiajing1126) [reply src/SelectInput.tsx]: Fixed with `prettier`

File: src/SelectInput.tsx [hunk 6]
```diff
@@ -82,7 +82,12 @@ function SelectInput<V>({
 	const previousItems = useRef<Array<Item<V>>>(items);
 
 	useEffect(() => {
-		if (!isEqual(previousItems.current, items)) {
+		if (
+			!isEqual(
+				previousItems.current.map(item => item.value),
```
Comment (vadimdemedes) [inline src/SelectInput.tsx:87]: Hm wait, I think that's incorrect. If `label` changes, list won't be re-rendered with the updated lable. Should it just be `isEqual(previousItems.current, items)`?

**Comment by vadimdemedes** [comment]
Thanks!

**Comment by vadimdemedes** [comment]
Released in https://github.com/vadimdemedes/ink-select-input/releases/tag/v4.2.0.
